content/app-dev/create-provider/cdd.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Contract-Driven Design"
+date: 2018-12-29T11:02:05+06:00
+weight: 6
+draft: false
+---
+
+The first thing we're going to need for our _payments service_ sample capability provider is a **contract**. This contract will describe the data types to be exchanged by the actor and provider, and the _supported operations_ that can be invoked on this contract. Our example may be a little simpler than a traditional payments system, but you should hopefully get a good idea for what this process looks like.
```

Comment:
The first thing we're going to need for our _payment service_ sample capability provider is a **contract**. A contract describes the data types Actors and Providers exchange as well as the _supported operations_ (methods? functions? allowed operations?) that can be invoked. 

Maybe consider dropping the qualifier about it being simpler than a traditional payments system? I feel like it interferes with the cadence of the docs. Do we want people reading, doing, feeling good about doing, asking questions about what they did and could potentially do differently in that order?